### ISSUE TITLE
Missing method to add RelationshipPattern as where on PatternComprehension

### ISSUE THREAD

**Issue by 3rrki** [neo4j-contrib/cypher-dsl#113]
Hello!

I am trying to construct a Cypher similar to below:

```
MATCH(foo:Foo) RETURN foo,[(foo)-[rel:FOOBAR]->(bar:Bar) WHERE (bar)-[:HAS]->(:ZZZ) |[rel,bar]];
```

But having some troubles to add the WHERE part

I tried something like this:

```
        var foo = Cypher.node("Foo").named("foo");
        var bar = Cypher.node("Bar").named("bar");
        var fooBar = foo.relationshipTo(bar, "FOOBAR").named("rel");
        var pc = Cypher.listBasedOn(fooBar)
                .where(bar.relationshipTo(Cypher.node("ZZZ"), "HAS"))
                .returning(fooBar, bar);
        var query =
                Cypher.match(foo).returning(foo.getRequiredSymbolicName(),  pc);
        System.out.println(Renderer.getDefaultRenderer().render(query.build()));
```

But the "where" method on the PatternComprehension does not accept a RelationshipPattern, only Conditions.

Is there a way to convert the Relationship to a Condition or is there another way to do this?

regards,

**Comment by michael-simons** [comment]
Hi!
Oh yes indeed, that's missing.

I can offer you the following workaround until the next patch:

```java
@Test
	void f() {
		var foo = Cypher.node("Foo").named("foo");
		var bar = Cypher.node("Bar").named("bar");
		var fooBar = foo.relationshipTo(bar, "FOOBAR").named("rel");
		var pc = Cypher.listBasedOn(fooBar)
			.where(Conditions.noCondition().and(bar.relationshipTo(Cypher.node("ZZZ"), "HAS")))
			.returning(fooBar, bar);
		var query =
			Cypher.match(foo).returning(foo.getRequiredSymbolicName(),  pc);
		System.out.println(Renderer.getDefaultRenderer().render(query.build()));
	}
```

Renders: `MATCH (foo:`Foo`) RETURN foo, [(foo)-[rel:`FOOBAR`]->(bar:`Bar`) WHERE (bar)-[:`HAS`]->(:`ZZZ`) | [rel, bar]]`

**Comment by 3rrki** [comment]
Hi

Thanks for that, it was more elegant than what I did (created a custom Condition class accepting a Relationship).

Regards,

**Comment by michael-simons** [comment]
No worries and thank you for reporting this. 

Implemented in e73afaa9ebd60496262084596811774ebb983f1e, will be out with the next patch.
